src/components/pages/OktaLoginForm.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button } from '@mui/material';
+import { useEffect, useState } from 'react';
+
+import { getCsrfToken } from '@/utils/csrf';
+
+interface Props {
+  path: string;
+}
+const OktaLoginForm: React.FC<Props> = ({ path }) => {
+  // fixme - csrf should probably be in a context or other shared state
+  const [csrf, setCsrf] = useState<string>();
+  useEffect(() => {
+    const token = getCsrfToken();
+    if (token) {
+      setCsrf(token);
+    }
+  }, []);
```

Comment:
I don't think we want to store the cookie in state, because it changes with every request. We only need to grab it when we are making an auth request.

So, this seems good. I don't think the effect hooks are _really_ necessary, because this component shouldn't be re-rendering. I think `const csrf = getCsrfToken();` would be just fine. Am I missing something?

Reply:
I encountered issues with missing/invalid csrf tokens in development. I suspect there's a timing issue; maybe since the api endpoints often takes some time to resolve, particularly on a cold start.

This useEffect is a bit of a placeholder / timing hack that seemed to make the form more reliable for me. You're right though, it's not what we'd want in production. I'll take it out.

My thought was that if the token is in state, we can update it as API calls resolve, allowing the login to be more robust against missing / invalid tokens. But maybe just reloading the page when the token is missing is sufficient